Social: Add a 'more info' link to the editor upgrade nudge.

pablinos opened the pull request:
#### Changes proposed in this Pull Request:

The upgrade link in the editor drops the customer directly in the checkout screen. This PR adds a more info link that takes them to the main jetpack.com pricing page, with the Jetpack Social plan details open in a modal.

If they checkout from here they will be redirected back to the editor.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

1202583289493814-as-1203337382116439


#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

#### Testing instructions:
~~This has highlighted an issue with the Jetpack redirects handling of query parameters. In order to test this properly you'll need to sandbox jetpack.com and apply D93475-code, until that or a similar fix is merged.~~

Although the Jetpack redirects issue still exists, we've changed the behaviour to work around it.

* Activate Jetpack Social on a test site, and ensure that the sharing limits are enabled and that there isn't a plan associated with the site.
* In the editor notice there is a 'More about...' link that opens a new tab with the pricing page on jetpack.com
* After purchasing a plan, you should be taken to the Jetpack Social admin page
* With Jetpack activated, the same process will take you to the default of the Jetpack recommendations admin page.

pablinos committed:
Social: Add a 'more info' link to the editor upgrade nudge.

The upgrade link in the editor drops the customer directly in the
checkout screen. This PR adds a more info link that takes them to the
main jetpack.com pricing page, with the Jetpack Social plan details open
in a modal.

If they checkout from here they will be redirected back to the editor.

gmjuhasz commented:
Nice work, the code looks good and tests as described!

I am just wondering if the learn more link should be opened in a new tab rather than on the current one. Usually these open in new tabs don't they? If the user just looks at the details and decides not to purchase, they can close the window and not have to click back a few times. What do you think?

Can we add a bit of space between the two sentences? The two links seem a bit cramped to me, but I could be wrong.
<img width="265" alt="CleanShot 2022-11-28 at 11 17 13@2x" src="https://user-images.githubusercontent.com/36671565/204252723-7a02f15d-1d3b-427d-96ba-bebb96beb26f.png">

pablinos commented:
> I am just wondering if the learn more link should be opened in a new tab 

Yes, this is a dilemma. The Upgrade link currently opens in the same window and then redirects you back to the editor, so this follows the same pattern, but I agree that 'more info' is the kind of thing that would normally open in a new tab and then close - or a modal like we orginally wanted!

The problem is that it's quite a bit more work. I don't think we have a way to keep people on a 'thank you' screen at the end of checkout, and we would need a way of refreshing state in the editor when focus is returned. It's not horrendous, but not as quick as this change.

> Can we add a bit of space between the two sentences?

Yes, I think we could do with updating the styling a little. I'll look at that.

pablinos committed:
Update the more info link to open in a pop-up.

pablinos commented:
I've got the pop-up working for the 'more info' link. If someone buys from there, they may end up on an odd page, in the same way as the current upgrade link, but I'm auto-saving the post now, so that should mean, whatever happens, that the user doesn't lose any work.

pablinos committed:
Update the more info link to open in a new tab

This also updates the redirect, so the user is sent back to a plugin
admin page after they have purcchased a plan.

pablinos committed:
changelog

gmjuhasz reviewed (dismissed):
This looks good to me and works well, nice job!

pablinos committed:
Merge branch 'trunk' into fix/social-upgrade-redirects

pablinos committed:
Fix package version

pablinos commented:
I lost the review! It should just need a final check over.

manzoorwanijk commented on projects/js-packages/publicize-components/src/components/form/index.js:
```diff
@@ -53,35 +56,78 @@ export default function PublicizeForm( {
+										}
+									) }
+									<br />
+									{ createInterpolateElement(
+										/* translators: %d is the number of shares remaining, moreLink is the link to find out more information about the plan */
+										__( '<moreLink>More about Jetpack Social</moreLink>.', 'jetpack' ),
+										{
+											moreLink: (
+												<a
+													className={ styles[ 'more-link' ] }
+													href={ getRedirectUrl( 'jetpack-social-block-editor-more-info', {
+														site: getSiteFragment(),
+														...( adminUrl
+															? { query: 'redirect_to=' + encodeURIComponent( adminUrl ) }
+															: {} ),
+													} ) }
+													target="_blank"
+													rel="noreferrer"
+													onClick={ autosaveAndRedirect }
+												/>
+											),
+										}
+									) }
```
Just noticed this in the code while trying to work on the sharing limit notice.

There is no need of `createInterpolateElement` here, it can simply be this to simplify the translation:

```jsx
<a
	className={ styles[ 'more-link' ] }
	href={ getRedirectUrl( 'jetpack-social-block-editor-more-info', {
		site: getSiteFragment(),
		...( adminUrl
			? { query: 'redirect_to=' + encodeURIComponent( adminUrl ) }
			: {} ),
	} ) }
	target="_blank"
	rel="noreferrer"
	onClick={ autosaveAndRedirect }
>
	{ __( 'More about Jetpack Social.', 'jetpack' ) }
</a>
```

pablinos replied:
Yes, good point! I think I copied it from somewhere and the text changed. Fancy creating a PR for this?

manzoorwanijk replied:
Here is the PR #31185

manzoorwanijk commented on projects/packages/publicize/src/class-share-limits.php:
```diff
@@ -95,7 +108,17 @@ public function render_classic_editor_notice() {
+				)
+			)
+		);
+		$more_link = sprintf(
+			/* translators: %s: link to find out more about the plan, and potentially upgrade. */
+			__( '<a href="%s" target="_blank">More about Jetpack Social</a>', 'jetpack-publicize-pkg' ),
+			Redirect::get_url(
+				'jetpack-social-basic-plan-block-editor',
+				array(
+					'query' => 'redirect_to=' . rawurlencode( $current_url ),
```
Same simplification here to re-use the translation:

```php
$more_link = sprintf(
	'<a href="%1$s" target="_blank">%2$s</a>',
	Redirect::get_url(
		'jetpack-social-basic-plan-block-editor',
		array(
			'query' => 'redirect_to=' . rawurlencode( $current_url ),
		)
	),
	__( 'More about Jetpack Social', 'jetpack-publicize-pkg' ),
);
```
